Troubleshooting guides cleanup (#7770)
kyma-project/kyma: docs/service-mesh/10-02-no-access.md

To fix this problem, restart the Pods of the Gateway.

1. List all available endpoints:

    ```bash
    kubectl get virtualservice --all-namespaces
    ```

2. Restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:

     ```bash
     kubectl delete pod -l app=istio-ingressgateway -n istio-system
     ```

> If the above solution doesn't work you need to change image of Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside of them. 

Reviewer: ```suggestion
If this solution doesn't work, you need to change the image of the Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside them. 
```

> 

Reviewer: Follow these steps:

1. Edit the Istio Ingress Gateway Deployment:

    ```bash
    kubectl scale --replicas 0 -n istio-system deploy/istio-ingressgateway
    kubectl edit deployment -n istio-system istio-ingressgateway
    kubectl scale --replicas 1 -n istio-system deploy/istio-ingressgateway
    ```
   

>    Find the `istio-proxy` container and delete the `-distroless` suffix.

Reviewer: ```suggestion
   2. Find the `istio-proxy` container and delete the `-distroless` suffix.
```

> 2. Check all ports used by the Istio Ingress Gateway:

Reviewer: ```suggestion
3. Check all ports used by the Istio Ingress Gateway:
```

>     kubectl scale --replicas 1 -n istio-system deploy/istio-ingressgateway

Reviewer 2: Hmm, why do you need to scale to 0 and then to 1? Just do kubectl edit directly. Afterward, you will have only one replica

    ```bash
    kubectl exec -ti -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system -o name) -c istio-proxy -- netstat -lptnu
    ```

> 3. If ports `80` and `443` are not used, restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:

Reviewer: ```suggestion
4. If ports `80` and `443` are not used, restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
```

> 

Reviewer 2: Hmm, previous steps mention to restart pod, if it doesn't help we modify image to use non distroless image which force another restart of pod, if mentioned command will show that port 443 is not opened what will another restart bring? IMO nothing :) In such case we need to find the root cause, mostly it is due to corrupted or missing certificate secret. Ingress will not open port 443 untill all mentioned secrets with certificates (They are defined on Gateways) are properly loaded by `istio-sds` container.

So we check logs from `istio-sds`container in istio-ingressgateway pod, look for errors + ensure secrets exists in istio-system namespace. if they exist check/validate  certificate if it's not for example string `aaaaa`. `istio-sds` in most of the cases report any problems it find with secrets referenced on Gateways (CRs) :)

  kubectl get virtualservice --all-namespaces
  ```

2. Restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
     ```
     kubectl delete pod -l app=istio-ingressgateway -n istio-system
     ```

If the above solution doesn't work you need to change image of Istio Ingress Gateway to allow further investigation. Kyma uses distroless Istio images which are more secure, but you cannot execute commands inside of them. 

1. Edit the Istio Ingress Gateway Deployment:

    ```

>    kubectl edit deployment -n istio-system istio-ingressgateway

Reviewer 3: ```suggestion
    ```bash
    kubectl scale --replicas 0 -n istio-system deploy/istio-ingressgateway
    kubectl edit deployment -n istio-system istio-ingressgateway
    kubectl scale --replicas 1 -n istio-system deploy/istio-ingressgateway
```

    ```
   

>    Find the `ingress-sds` container and delete the `-distroless` suffix.

Reviewer 3: ```suggestion
   Find the `istio-proxy` container and delete the `-distroless` suffix.
```

2. Check all ports used by the Istio Ingress Gateway:
  ```

>   kubectl exec -t -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system | grep "istio-ingressgateway" | awk '{print $1}') -- netstat -lptnu

Reviewer 3: ```suggestion
  ```bash
  kubectl exec -ti -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system -o name) -c istio-proxy -- netstat -lptnu
```
